Cython/Compiler/Nodes.py

```diff
@@ -966,6 +966,9 @@ def inject_type_from_annotations(self, env):
         if base_type is not None:
             self.base_type = base_type
 
+        if arg_type is py_object_type:
+            # exclude ": object" from the None check - None is an generic object
+            self.or_none = True
```

Comment:
I was considering whether there should be an `elif` following, but we also have to allow `Optional[object]` (because why not). But we could move this second, after the `Optional` handling and before the `None` default handling, what do you think?

Reply:
I looked at it a bit and wasn't sure whether it mattered so much. It definitely needs to be before the `None` default handling, but the rest of the order isn't so important.

`Optional[object]` should end up fine with any of the sensible versions I think - it won't pass the `arg_type is py_object_type` check, but everything necessary will be handled by the `Optional` check.

Essentially, if you want to re-arrange it a little for clarity then that's fine!

Reply:
Looks like I lost track of the fact that the only reason why this wasn't merged yet was a nit-pick. Sorry for that.
I rearranged the if-clauses, though. I think it's a little clearer that way what the different cases are.